src/widgets/MultiSelectList.js

```diff
@@ -0,0 +1,235 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2016
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { SearchBar } from 'react-native-ui-components';
+import { Dimensions, FlatList, StyleSheet, View } from 'react-native';
+import { complement } from 'set-manipulator';
+import globalStyles, { APP_FONT_FAMILY, SUSSOL_ORANGE } from '../globalStyles';
+import { generalStrings, buttonStrings } from '../localization';
+import { OnePressButton } from '.';
+import { ResultRow } from './ResultRow';
+
+/**
+ * A search bar that autocompletes from the options passed in, and allows any of
+ * the dropdown options to be selected. Will gravefully handle null values
+ * by using an empty array of searchable objects.
+ * @prop  {array}     options         The options to select from
+ * @prop  {function}  onSelect        A function taking the selected option as a parameter
+ * @prop  {string}    queryString     The query to filter the options by, where $0 will
+ *                                    be replaced by the user's current search
+ *                                    e.g. 'name BEGINSWITH[c] $0 OR code BEGINSWITH[c] $0'
+ * @prop  {string}    placeholderText The text to initially display in the search bar
+ */
+class MultiSelectList extends React.PureComponent {
```

Comment:
would be good for this to be a functional component using newer hooks syntax

Reply:
Resolved with d372606